stellar-contract-env-host/src/test.rs

```diff
@@ -1173,3 +1168,197 @@ fn binary_xdr_roundtrip() -> Result<(), HostError> {
+        );
+    }
+    // and
+    {
+        let obj_res = host.bigint_and(obj_a, obj_b)?;
+        let obj_ref = host.bigint_from_i64(a as i64 & b)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+    }
+    // or
+    {
+        let obj_res = host.bigint_or(obj_a, obj_b)?;
+        let obj_ref = host.bigint_from_i64(a as i64 | b)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+    }
+    // xor
+    {
+        let obj_res = host.bigint_xor(obj_a, obj_b)?;
+        let obj_ref = host.bigint_from_i64(a as i64 ^ b)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+    }
+    // shl
+    {
+        let obj_res = host.bigint_shl(obj_a, 5)?;
+        let obj_ref = host.bigint_from_u64(a << 5)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+    }
+    // shr
+    {
+        let obj_res = host.bigint_shr(obj_a, 5)?;
+        let obj_ref = host.bigint_from_u64(a >> 5)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+    }
+    // cmp
+    {
+        use std::cmp::Ordering;
+        let ord_greater: i32 = host.bigint_cmp(obj_a, obj_b)?.try_into()?;
+        let ord_less: i32 = host.bigint_cmp(obj_b, obj_a)?.try_into()?;
+        let obj3 = host.bigint_from_u64(a)?;
+        let ord_equal: i32 = host.bigint_cmp(obj_a, obj3)?.try_into()?;
+        assert_eq!(ord_greater, Ordering::Greater as i32);
+        assert_eq!(ord_less, Ordering::Less as i32);
+        assert_eq!(ord_equal, Ordering::Equal as i32);
+    }
+    // is zero
+    {
+        let f = RawVal::from_bool(false);
+        let t = RawVal::from_bool(true);
+        assert_eq!(host.bigint_is_zero(obj_a)?.get_payload(), f.get_payload());
+        assert_eq!(host.bigint_is_zero(obj_b)?.get_payload(), f.get_payload());
+        assert_eq!(host.bigint_is_zero(obj_0)?.get_payload(), t.get_payload());
+    }
+    // neg
+    {
+        let obj_res = host.bigint_neg(obj_b)?;
+        let obj_ref = host.bigint_from_i64(-b)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+        assert_eq!(
+            host.obj_cmp(host.bigint_neg(obj_res)?.into(), obj_b.into())?,
+            0
+        );
+    }
+    // not
+    {
+        let obj_res = host.bigint_not(obj_b)?;
+        let obj_ref = host.bigint_from_i64(!b)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+        assert_eq!(
+            host.obj_cmp(host.bigint_not(obj_res)?.into(), obj_b.into())?,
+            0
+        );
+    }
+    // gcd
+    {
+        let obj_res = host.bigint_gcd(obj_a, obj_b)?;
+        let obj_ref = host.bigint_from_i64(10)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_ref.into())?, 0);
+        // gcd by 0 is self
+        let obj_res = host.bigint_gcd(obj_a, obj_0)?;
+        assert_eq!(host.obj_cmp(obj_res.into(), obj_a.into())?, 0);
+    }
+    // lcm
+    {
+        // let bi_c = bi_a.lcm(&bi_b);
```

Comment:
Nitpick: I assume you meant to remove all of these comments (in lcm, pow, pow_mod, sqrt, and bits)?